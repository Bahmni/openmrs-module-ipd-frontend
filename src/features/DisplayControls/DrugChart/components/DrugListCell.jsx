import React from "react";
import PropTypes from "prop-types";
import Clock from "../../../../icons/clock.svg";
import "../styles/DrugListCell.scss";
import { TooltipCarbon } from "bahmni-carbon-ui";
import NoteIcon from "../../../../icons/note.svg";
import DisplayTags from "../../../../components/DisplayTags/DisplayTags";

export default function DrugListCell(props) {
  const { drugInfo } = props;
  const {
    drugName,
    dosage,
    doseType,
    drugRoute,
    duration,
    administrationInfo,
    dosingInstructions,
    dosingTagInfo,
  } = drugInfo;

  let parsedDosingInstructions;
  if (dosingInstructions !== null && dosingInstructions !== undefined) {
    parsedDosingInstructions = JSON.parse(dosingInstructions);
  }

  const toolTipContent = (
    <div>
      Instructions:&nbsp;{parsedDosingInstructions.instructions}
      <br />
      {parsedDosingInstructions.additionalInstructions && (
        <>
          <div className="tooltip-content-separater" />
          Additional Instructions:&nbsp;
          {parsedDosingInstructions.additionalInstructions}
        </>
      )}
    </div>
  );
  const drugNameText = (
    <div className="drug-name-container">
      <div className={"drug-chart-drug-name"}>{drugName}</div>
      {dosingInstructions && (
        <TooltipCarbon icon={() => icon} content={toolTipContent} />
      )}
      <div className="drug-name-cell">
        <DisplayTags drugOrder={dosingTagInfo} />
      </div>
    </div>
  );
  const icon = (
    <div className="note-icon-container">
      <NoteIcon />
    </div>
  );
  return (
    <td>
      {drugNameText}

      <div className={"dosage"}>
        <span>{dosage}</span>
        {doseType && <span>&nbsp;-&nbsp;{doseType}</span>}
        <span>&nbsp;-&nbsp;{drugRoute}</span>
        {duration && <span>&nbsp;-&nbsp;{duration}</span>}
      </div>
      {administrationInfo.length >= 1 && (
        <div style={{ display: "flex", alignItems: "center", gap: "5px" }}>
          <Clock />
          {administrationInfo.map((adminInfo, index) => {
            if (adminInfo.kind === "Administered-Late") {
              return (
                <span style={{ color: "#da1e28" }} key={index}>
                  {adminInfo.time}
                  {index !== administrationInfo.length - 1 && (
                    <span style={{ color: "#525252" }}>,</span>
                  )}
                </span>
              );
            } else {
              return (
                <span style={{ color: "#525252" }} key={index}>
                  {adminInfo.time}
                  {index !== administrationInfo.length - 1 && <span>,</span>}
                </span>
              );
            }
          })}
        </div>
      )}
    </td>
  );
}
DrugListCell.propTypes = {
  drugInfo: PropTypes.object.isRequired,
};
