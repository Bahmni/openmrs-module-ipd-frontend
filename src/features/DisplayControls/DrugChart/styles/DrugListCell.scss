.drug-chart-drug-name {
  max-width: 85%;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-weight: 500;
  color: #393939;
  font-style: normal;
}
.drug-name-container {
  display: flex;
}
.dosage {
  font-size: 12px;
  font-weight: 400;
  color: #393939;
  > * {
    display: inline-block;
  }
}
.note-icon-container {
  margin-left: 5px;
}
.tooltip-content-separater {
  height: 1px;
  background-color: #bbb8b8;
  width: 100%;
  margin-top: 10px;
  margin-bottom: 10px;
}
