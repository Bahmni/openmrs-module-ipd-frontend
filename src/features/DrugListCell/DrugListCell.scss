.drug-name {
  max-width: 85%;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  font-weight: 500;
  color: #393939;
  font-style: normal;
}
.dosage {
  font-size: 12px;
  font-weight: 400;
  color: #393939;
  > * {
    display: inline-block;
  }
}
