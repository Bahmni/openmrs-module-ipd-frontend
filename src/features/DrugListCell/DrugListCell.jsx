import React from "react";
import PropTypes from "prop-types";
import Clock from "../../icons/clock.svg";
import "./DrugListCell.scss";
// import { TooltipDefinitionCarbon } from "bahmni-carbon-ui";

export default function DrugListCell(props) {
  const { drugInfo } = props;
  const {
    drugName,
    dosage,
    doseType,
    drugRoute,
    duration,
    administrationInfo,
  } = drugInfo;
  const drugNameText = <div className={"drug-name"}>{drugName}</div>;
  //TODO: Add tooltip for drug name
  return (
    <td>
      {/*<TooltipDefinitionCarbon tooltipText={drugName} content={drugNameText}/>*/}
      {drugNameText}
      <div className={"dosage"}>
        <span>{dosage}</span>
        {doseType && <span>&nbsp;-&nbsp;{doseType}</span>}
        <span>&nbsp;-&nbsp;{drugRoute}</span>
        {duration && <span>&nbsp;-&nbsp;{duration}</span>}
      </div>
      {administrationInfo.length >= 1 && (
        <div style={{ display: "flex", alignItems: "center", gap: "5px" }}>
          <Clock />
          {administrationInfo.map((adminInfo, index) => {
            if (adminInfo.kind === "Administered-Late") {
              return (
                <span style={{ color: "#da1e28" }} key={index}>
                  {adminInfo.time}
                  {index !== administrationInfo.length - 1 && (
                    <span style={{ color: "#525252" }}>,</span>
                  )}
                </span>
              );
            } else {
              return (
                <span style={{ color: "#525252" }} key={index}>
                  {adminInfo.time}
                  {index !== administrationInfo.length - 1 && <span>,</span>}
                </span>
              );
            }
          })}
        </div>
      )}
    </td>
  );
}
DrugListCell.propTypes = {
  drugInfo: PropTypes.object.isRequired,
};
