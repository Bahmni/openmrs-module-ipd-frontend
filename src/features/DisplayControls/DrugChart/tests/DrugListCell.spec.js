import React from "react";
import { render } from "@testing-library/react";
import DrugListCell from "../components/DrugListCell";
import {
  testDrugInfo,
  testDrugInfoWithAdministeredLateStatus,
  testDrugInfoWithAdministeredStatus,
} from "./DrugListCellMockData";

const MockTooltipCarbon = jest.fn();
jest.mock("../../../../icons/note.svg");

jest.mock("bahmni-carbon-ui", () => {
  return {
    TooltipCarbon: (props) => {
      MockTooltipCarbon(props);
      return <div>TooltipCarbon</div>;
    },
  };
});

describe("DrugListCell", () => {
  it("should  match snapshot", () => {
    const { container } = render(<DrugListCell drugInfo={testDrugInfo} />);
    expect(container).toMatchSnapshot();
  });

  it("should  match snapshot for for drug cell with admin info status as Administered-Late", () => {
    const { container } = render(
      <DrugListCell drugInfo={testDrugInfoWithAdministeredLateStatus} />
    );
    expect(container).toMatchSnapshot();
  });

  it("should  match snapshot for drug cell with admin info status as Administered", () => {
    const { container } = render(
      <DrugListCell drugInfo={testDrugInfoWithAdministeredStatus} />
    );
    expect(container).toMatchSnapshot();
  });
});
