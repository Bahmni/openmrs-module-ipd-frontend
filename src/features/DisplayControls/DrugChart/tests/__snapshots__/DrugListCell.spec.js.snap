// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DrugListCell should  match snapshot 1`] = `
<div>
  <td>
    <div
      class="drug-name"
    >
      Suxamethonium Chloride 100 mg/2 mL Ampoule (50 mg/mL )
    </div>
    <div
      class="dosage"
    >
      <span>
        2
      </span>
      <span>
         - 
        mg
      </span>
      <span>
         - 
        Intravenous
      </span>
      <span>
         - 
        4 Day(s)
      </span>
    </div>
  </td>
</div>
`;

exports[`DrugListCell should  match snapshot for drug cell with admin info status as Administered 1`] = `
<div>
  <td>
    <div
      class="drug-name"
    >
      Enalapril 5 mg Tablet
    </div>
    <div
      class="dosage"
    >
      <span>
        1
      </span>
      <span>
         - 
        Tablet(s)
      </span>
      <span>
         - 
        Oral
      </span>
      <span>
         - 
        4 Day(s)
      </span>
    </div>
    <div
      style="display: flex; align-items: center; gap: 5px;"
    >
      <test-file-stub />
      <span
        style="color: rgb(82, 82, 82);"
      >
        8.45
      </span>
    </div>
  </td>
</div>
`;

exports[`DrugListCell should  match snapshot for for drug cell with admin info status as Administered-Late 1`] = `
<div>
  <td>
    <div
      class="drug-name"
    >
      Enalapril 5 mg Tablet
    </div>
    <div
      class="dosage"
    >
      <span>
        1
      </span>
      <span>
         - 
        Tablet(s)
      </span>
      <span>
         - 
        Oral
      </span>
      <span>
         - 
        4 Day(s)
      </span>
    </div>
    <div
      style="display: flex; align-items: center; gap: 5px;"
    >
      <test-file-stub />
      <span
        style="color: rgb(218, 30, 40);"
      >
        8.45
      </span>
    </div>
  </td>
</div>
`;
